### PR TITLE
init project

### PR DESCRIPTION
**Pull request by trinhdinhhiep**
- init project
- config ESLint, Prettier, husky

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+dist
+dist-ssr
+*.local
+
+# Editor directories and files
+.vscode/*
+!.vscode/extensions.json
+.idea
+.DS_Store
+*.suo
```
Comment (tuev) [inline .gitignore]: ??

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+*.ntvs*
```
Comment (tuev) [inline .gitignore]: remove all unused type

File: .husky/pre-commit [hunk 3]
```diff
@@ -0,0 +1 @@
+yarn check:all
```
Comment (tuev) [inline .husky/pre-commit]: should add linting for pre-commit in husky

File: .prettierrc [hunk 4]
```diff
@@ -0,0 +1,11 @@
+{
+  "arrowParens": "always",
+  "semi": false,
+  "trailingComma": "none",
```
Comment (tuev) [inline .prettierrc]: should use trailing comma

File: eslint.config.js [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+      "react-refresh": reactRefresh,
+      prettier: eslintPluginPrettier,
+    },
+    rules: {
+      ...reactHooks.configs.recommended.rules,
+      "react-refresh/only-export-components": [
+        "warn",
+        { allowConstantExport: true },
+      ],
+      "prettier/prettier": [
```
Comment (tuev) [inline eslint.config.js]: duplicate with pretierrc

File: src/index.css [hunk 6]
```diff
@@ -0,0 +1,68 @@
+:root {
```
Comment (tuev) [inline src/index.css:3]: need add normalizecss

File: tsconfig.app.json [hunk 7]
```diff
@@ -0,0 +1,25 @@
+{
```
Comment (tuev) [inline tsconfig.app.json:1]: why do we need tsconfig.app and tsconfig.node

File: .husky/pre-commit (hunk 3 above)
Comment (trinhdinhhiep) [reply .husky/pre-commit]: yarn check:all includes: linting check, Prettier check, and type check.

**Comment by vuquangit** [comment]
Change npm to pnpm

File: yarn.lock
Comment (vuquangit) [inline yarn.lock:1]: remove this file

File: .husky/pre-commit (hunk 3 above)
Comment (mrbin264) [reply .husky/pre-commit]: Why is "yarn check:all" being used when the project has switched to pnpm?

**Review by tuev — CHANGES_REQUESTED** [review]

File: tsconfig.app.json (hunk 7 above)
Comment (trinhdinhhiep) [reply tsconfig.app.json:1]: Vite creates tsconfig.app and tsconfig.node to optimize TypeScript compilation for browser and Node.js environments, respectively. This separation prevents conflicts and improves build performance.

File: src/index.css (hunk 6 above)
Comment (trinhdinhhiep) [reply src/index.css:3]: done

**Review by tuev — APPROVED** [review]
LGTM

**Review by vuquangit — APPROVED** [review]

**Review by mrbin264 — APPROVED** [review]

**Pull request merged**